packages/dev/core/src/Meshes/meshUtils.ts

```diff
@@ -0,0 +1,209 @@
+    animationGroup: Nullable<AnimationGroup> = null,
+    animationStep = 1 / 6
+): Array<{ minimum: Vector3; maximum: Vector3 }> {
+    // Local vector to avoid allocations.
+    const position = TmpVectors.Vector3[0];
+
+    const meshExtents = new Map<number, { minimum: Vector3; maximum: Vector3 }>();
+    const skinnedMeshExtents = new Map<number, Map<number, { minimum: Vector3; maximum: Vector3 }>>();
+
+    // Compute the non-skinned and skinned mesh extents.
+    for (const mesh of meshes) {
+        const positions = mesh.getVerticesData(VertexBuffer.PositionKind);
+        if (!positions) {
+            continue;
+        }
+
+        // Initialize min/max positions with the original positions.
+        const numVertices = mesh.getTotalVertices();
+        const minPositions = new Array<Vector3>(numVertices);
+        const maxPositions = new Array<Vector3>(numVertices);
+        for (let i = 0, j = 0; i < numVertices; i++, j += 3) {
+            position.set(positions[j], positions[j + 1], positions[j + 2]);
+            minPositions[i] = position.clone();
+            maxPositions[i] = position.clone();
+        }
+
+        // Apply morph targets to the min/max positions.
+        const morphTargetManager = mesh.morphTargetManager;
+        if (morphTargetManager) {
+            for (let targetIndex = 0; targetIndex < morphTargetManager.numTargets; ++targetIndex) {
+                const target = morphTargetManager.getTarget(targetIndex);
+                const positions = target.getPositions();
+                if (positions) {
+                    for (let i = 0, j = 0; i < numVertices; i++, j += 3) {
+                        position.set(positions[j], positions[j + 1], positions[j + 2]);
+                        minPositions[i].minimizeInPlace(position);
+                        maxPositions[i].maximizeInPlace(position);
+                    }
+                }
+            }
+        }
+
+        // Compute extent per mesh.
+        const skeleton = mesh.skeleton;
+        const weights = skeleton ? mesh.getVerticesData(VertexBuffer.MatricesWeightsKind) : null;
+        const indices = skeleton ? mesh.getVerticesData(VertexBuffer.MatricesIndicesKind) : null;
+        if (weights && indices) {
+            // Compute extent per bone for skinned meshes.
+            const needsExtra = mesh.numBoneInfluencers > 4;
+            const weightsExtra = needsExtra ? mesh.getVerticesData(VertexBuffer.MatricesWeightsExtraKind) : null;
+            const indicesExtra = needsExtra ? mesh.getVerticesData(VertexBuffer.MatricesIndicesExtraKind) : null;
+
+            const perBoneExtents = skinnedMeshExtents.get(mesh.uniqueId) || new Map<number, { minimum: Vector3; maximum: Vector3 }>();
+            skinnedMeshExtents.set(mesh.uniqueId, perBoneExtents);
+
+            for (let i = 0, j = 0; i < numVertices; i++, j += 4) {
+                const updateExtents = (weights: FloatArray, indices: FloatArray): void => {

```

Comment:
if you define this function outside of the for loop and take `i` as a parameter you can save some perf